File: flink/src/main/scala/pl/mproch/streaming/flink/Message.scala
package pl.mproch.streaming.flink

case class Message(userId: String, topic: String, rate: Int, time: Long) {
  def withUser(user: Option[User]) = MessageWithUser(user, topic, rate, time)
}

case class User(userId: String, rank: Int, age: Int)

case class MessageWithUser(user: Option[User], topic: String, rate: Int, time: Long)


case class MessagesByUser(userId: String, count: Int)

case class TimedMessagesByUser(userId: String, count: Int, startTime: Long)

File: common/src/main/java/pl/mproch/streaming/model/Message.java
package pl.mproch.streaming.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Message {

    private long time;

    private String userId;

    private String topic;

    private int rate;

    private User user;

    public Message(long time, String userId, String topic, int rate) {
        this.time = time;
        this.userId = userId;
        this.topic = topic;
        this.rate = rate;
    }

    public Message withUser(User user) {
        return new Message(time, userId, topic, rate, user);

    }

}
